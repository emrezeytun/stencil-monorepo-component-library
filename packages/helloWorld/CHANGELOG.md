# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 1.0.2 (2025-08-24)


### Bug Fixes

* **button:** update button text and build outputs ([ea2c763](https://github.com/your-org/arcadia-components/commit/ea2c7633a7e06b930e71d9d40410eed19d3fa71b))


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))





## 1.0.1 (2025-08-24)


### Bug Fixes

* **button:** update button text and build outputs ([ea2c763](https://github.com/your-org/arcadia-components/commit/ea2c7633a7e06b930e71d9d40410eed19d3fa71b))


### Features

* setup [@arcadia-components](https://github.com/arcadia-components) organization with React, Vue, Angular bindings ([8cb0519](https://github.com/your-org/arcadia-components/commit/8cb0519888418c79da209e1b2807e05e1dc31cc6))
