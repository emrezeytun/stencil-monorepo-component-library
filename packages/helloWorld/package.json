{
  "name": "@comp-test/hello-world",
  "version": "1.0.0",
  "description": "Simple Hello World component built with Stencil",
  "main": "dist/index.cjs.js",
  "module": "dist/index.js",
  "types": "dist/types/index.d.ts",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "unpkg": "dist/hello-world/hello-world.esm.js",
  "exports": {
    ".": {
      "import": "./dist/hello-world/hello-world.esm.js",
      "require": "./dist/hello-world/hello-world.cjs.js",
      "types": "./dist/types/index.d.ts"
    },
    "./hello-world": {
      "import": "./dist/components/hello-world.js",
      "types": "./dist/components/hello-world.d.ts"
    },
    "./loader": {
      "import": "./loader/index.js",
      "require": "./loader/index.cjs",
      "types": "./loader/index.d.ts"
    },
    "./react": {
      "import": "./bindings/react/index.js",
      "types": "./bindings/react/index.d.ts"
    },
    "./vue": {
      "import": "./bindings/vue/index.js",
      "types": "./bindings/vue/index.d.ts"
    },
    "./angular": {
      "import": "./bindings/angular/index.js",
      "types": "./bindings/angular/index.d.ts"
    }
  },
  "files": [
    "dist/",
    "loader/",
    "bindings/"
  ],
  "scripts": {
    "build": "stencil build",
    "start": "stencil build --dev --watch --serve",
    "test": "stencil test --spec",
    "test.watch": "stencil test --spec --watchAll",
    "generate": "stencil generate",
    "storybook": "storybook dev -p 6008",
    "build-storybook": "storybook build"
  },
  "keywords": [
    "stencil",
    "web-components",
    "hello-world",
    "ui",
    "component"
  ],
  "author": "Your Name",
  "license": "MIT",
  "devDependencies": {
    "@stencil/angular-output-target": "^1.1.1",
    "@stencil/core": "^4.27.1",
    "@stencil/react-output-target": "^1.2.0",
    "@stencil/vue-output-target": "^0.11.8",
    "@storybook/addon-essentials": "^7.6.10",
    "@storybook/html-vite": "^7.6.10",
    "@types/jest": "^29.5.14",
    "@types/node": "^22.13.5",
    "jest": "^29.7.0",
    "jest-cli": "^29.7.0",
    "puppeteer": "^24.3.0",
    "storybook": "^7.6.10",
    "typescript": "^5.3.0"
  }
}
