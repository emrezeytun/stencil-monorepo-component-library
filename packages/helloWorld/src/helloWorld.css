.hello-world {
  padding: 2rem;
  text-align: center;
  font-family: Arial, sans-serif;
  border: 2px solid #007bff;
  border-radius: 8px;
  background: linear-gradient(135deg, #f8f9fa 0%, #e9ecef 100%);
}

.hello-world h1 {
  color: #007bff;
  margin: 0 0 1rem 0;
  font-size: 2rem;
}

.hello-world p {
  color: #495057;
  margin: 0;
  font-size: 1.1rem;
}
