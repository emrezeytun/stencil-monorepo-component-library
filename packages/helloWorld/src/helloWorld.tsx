import { Component, h } from '@stencil/core';

@Component({
  tag: 'hello-world',
  styleUrl: 'helloWorld.css',
  shadow: true,
})
export class HelloWorld {
  render() {
    return (
      <div class="hello-world">
        <h1>ğŸš€ Hot Reload Test! HMR Working!</h1>
        <p>Bu deÄŸiÅŸiklik build almadan gÃ¶rÃ¼nmeli!</p>
      </div>
    );
  }
}
