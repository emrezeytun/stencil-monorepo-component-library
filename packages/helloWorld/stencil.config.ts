import { Config } from '@stencil/core';
import { reactOutputTarget } from '@stencil/react-output-target';
import { vueOutputTarget } from '@stencil/vue-output-target';
import { angularOutputTarget } from '@stencil/angular-output-target';

const distTarget = {
  type: 'dist-custom-elements',
  dir: 'dist/components',
  customElementsExportBehavior: 'single-export-module'
};

export const config: Config = {
  namespace: 'hello-world',
  taskQueue: 'async',
  srcDir: 'src',
  outputTargets: [
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
    {
      type: 'dist-custom-elements',
      customElementsExportBehavior: 'bundle',
      externalRuntime: false,
    },
    // React bindings için gerekli custom elements
    distTarget,

    {
      type: 'docs-readme',
      footer: 'Built with ❤️ using Stencil'
    },
    {
      type: 'www',
      serviceWorker: null,
    }
  ],
  testing: {
    browserHeadless: "shell",
  },
  devServer: {
    reloadStrategy: 'pageReload',
    port: 3334,
    openBrowser: false
  },
};
