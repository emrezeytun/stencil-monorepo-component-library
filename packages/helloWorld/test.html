<!DOCTYPE html>
<html>
<head>
  <title>HelloWorld Test</title>
  <script type="module" src="./loader/index.js"></script>
</head>
<body>
  <h1>Manual Test</h1>
  <hello-world></hello-world>
  
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      console.log('DOM loaded');
      console.log('Custom elements registry:', window.customElements);
      console.log('hello-world defined:', window.customElements.get('hello-world'));
      
      setTimeout(() => {
        const hwElement = document.querySelector('hello-world');
        console.log('HelloWorld element:', hwElement);
        console.log('Has shadow root:', !!hwElement.shadowRoot);
      }, 2000);
    });
  </script>
</body>
</html>
