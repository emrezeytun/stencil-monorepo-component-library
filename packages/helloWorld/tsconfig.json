{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "allowUnreachableCode": false,
    "declaration": false,
    "experimentalDecorators": true,
    "lib": [
      "dom",
      "es6"
    ],
    "moduleResolution": "node",
    "module": "esnext",
    "target": "es2017",
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "jsx": "react",
    "jsxFactory": "h",
    "jsxFragmentFactory": "Fragment",
    "strict": true,
    "skipLibCheck": false,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules",
    "dist",
    "www"
  ]
}
