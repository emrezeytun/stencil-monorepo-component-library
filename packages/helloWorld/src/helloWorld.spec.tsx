import { newSpecPage } from '@stencil/core/testing';
import { HelloWorld } from './helloWorld';

describe('hello-world', () => {
  it('renders without crashing (smoke test)', async () => {
    const page = await newSpecPage({
      components: [HelloWorld],
      html: `<hello-world></hello-world>`,
    });
    
    // Basic render check
    expect(page.root).toBeTruthy();
    expect(page.root?.tagName).toBe('HELLO-WORLD');
  });

  it('renders with correct content', async () => {
    const page = await newSpecPage({
      components: [HelloWorld],
      html: `<hello-world></hello-world>`,
    });
    
    // Check shadow DOM content
    const shadowRoot = page.root?.shadowRoot;
    expect(shadowRoot).toBeTruthy();
    
    const h1 = shadowRoot?.querySelector('h1');
    expect(h1).toBeTruthy();
    expect(h1?.textContent).toContain('HMR Test 2024');
    
    const p = shadowRoot?.querySelector('p');
    expect(p).toBeTruthy();
    expect(p?.textContent).toContain('aaaaa bbbb');
  });

  it('has correct CSS classes', async () => {
    const page = await newSpecPage({
      components: [HelloWorld],
      html: `<hello-world></hello-world>`,
    });
    
    const div = page.root?.shadowRoot?.querySelector('div');
    expect(div).toHaveClass('hello-world');
  });

  it('matches snapshot', async () => {
    const page = await newSpecPage({
      components: [HelloWorld],
      html: `<hello-world></hello-world>`,
    });
    
    expect(page.root).toMatchSnapshot();
  });
});
