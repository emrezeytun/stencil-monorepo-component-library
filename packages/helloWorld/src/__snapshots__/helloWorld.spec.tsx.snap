// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello-world matches snapshot 1`] = `
<hello-world>
  <template shadowrootmode="open">
    <div class="hello-world">
      <h1>
        🔥 UPDATED! HMR Test 2024! tttt abc abc 222
      </h1>
      <p>
        Bu değişiklik otomatik görünmeli - Test! 5555 🎯 ⚡ aaaaa bbbb!
      </p>
    </div>
  </template>
</hello-world>
`;
